tls/s2n_server_hello_retry.c

```diff
@@ -69,5 +68,34 @@ int s2n_server_hello_retry_send(struct s2n_connection *conn)
 
 int s2n_server_hello_retry_recv(struct s2n_connection *conn)
 {
-    S2N_ERROR(S2N_ERR_BAD_MESSAGE);
+    notnull_check(conn);
+
+    /* If the client receives a second HelloRetryRequest in the same connection, 
+    it MUST abort handshake. */
+    if (conn->handshake.client_hrr_received) {
```

Comment:
Is it possible to check the handshake type here, instead of the `client_hrr_received` flag? On the first pass, the client shouldn't be operating with a HELLO_RETRY_REQUEST type. So if it is present, then something is wrong.

Reply:
Would this be in the INITIAL state when first called? I could add this check as a replacement
```
    if (!strcasecmp(s2n_connection_get_handshake_type_name(conn), "HELLO_RETRY_REQUEST")){
        S2N_ERROR(S2N_ERR_BAD_MESSAGE);
    }
```

Reply:
Actually, `when s2n_server_hello_retry_recv` is invoked the first time we would have already parsed the client_hello extension, that would mean that `s2n_set_hello_retry_required()`  is called and the handshake is updated to `HELLO_RETRY_REQUEST`. 

From offline discussion:  We are keeping the flag as there isn't any other clever way to check for a server hello retry received by the client in the same connection.

Reply:
Offline discussion, checking the handshake does not work because HRR state is set during ClientHello.